common/dllserver/thorplugin.cpp

```diff
@@ -235,12 +235,64 @@ extern bool getResourceFromFile(const char *filename, MemoryBuffer &data, const
 
 //-------------------------------------------------------------------------------------------------------------------
 
+class ManifestFileList : public MappingBase
+{
+    StringArray filenames;
+    StringAttr type;
+    StringAttr dir;
+    void recursiveRemoveDirectory(const char *fullPath)
+    {
+        if (rmdir(fullPath) == 0 && !streq(fullPath, dir))
+        {
+            StringBuffer head;
+            splitFilename(fullPath, &head, &head, NULL, NULL);
+            if (head.length() > 1)
+            {
+                head.setLength(head.length()-1);
+                recursiveRemoveDirectory(head);
+            }
+        }
+    }
+    void removeFileAndEmptyParents(const char *fullFileName)
+    {
+        remove(fullFileName);
+        StringBuffer path;
+        splitFilename(fullFileName, &path, &path, NULL, NULL);
+        if (path.length() > 1)
+        {
+            path.setLength(path.length()-1);
+            recursiveRemoveDirectory(path.str());
+        }
+    }
+
+public:
+    ManifestFileList(const char *_type, const char *_dir) : type(_type), dir(_dir) {}
+    ~ManifestFileList()
+    {
+        ForEachItemIn(idx, filenames)
+        {
+            removeFileAndEmptyParents(filenames.item(idx));
+        }
+        rmdir(dir);  // If the specified temporary directory is now empty, remove it.
+    }
+    void append(const char *filename)
+    {
+        filenames.append(filename);
```

Comment:
Is it worth checking that the prefix of the filename matches dir? otherwise I might be concerned about recursiveRemoveDirectory walking too far